BD-3955 Add troubleshooting page for SAML/SSO setup (#9086)
braze-inc/braze-docs: _docs/_user_guide/administrative/access_braze/single_sign_on/set_up.md

Select **Export**. For **Select cookie-filter profile**, select **None**. Then, select **Export**. This will generate a JSON file that you can send to Braze Support for further troubleshooting.

!["Export SAML-trace preferences" menu with option "None" selected.]({% image_buster /assets/img/export_saml_trace_preferences.png %})

> ## Troubleshooting

Reviewer: side note as i continue reviewing: it may make more sense to rephrase headings to be similar to what a user would search on the web so it would do better with SEO.

not sure if there's some pre-existing guidelines/best practices out there for this kind of thing (since troubleshooting isn't in diataxis), but may want to checkout how stripe and microsoft handles it for inspo?

@rachel-feinberg

Author: @paulzs Do some of these troubleshooting items have respective errors?

Reviewer 2: Hey @rachel-feinberg, Emily and Bre recently did some error mapping and it can be found here: https://docs.google.com/document/d/1RwLYLTgVgLK_9ouQKnAIGD3fAimx09nTGXHSlQQSWTc/edit?tab=t.0

-----END CERTIFICATE-----
```

### The user has an invalid session token

Have the affected user [clear their browser's cache and cookies](https://its.uiowa.edu/services/how-clear-cache-and-cookies-your-web-browser), and then try to log in with SAML SSO again.

### Your company hasn't set the RelayState

To authenticate using your identity provider (IdP), your company needs to set your RelayState in your IdP management system.

1. In Braze, go to **Settings** > **APIs and Identifiers**.
2. In the **API Keys** tab, select the **Create API key** button.
3. Enter a key name in the field **API key name**.
4. Extend the **SSO** dropdown under **Permissions** and check **sso.saml.login**.<br><br>![The "Permissions" section with sso.saml.login checked.]({% image_buster /assets/img/relaystate_troubleshoot.png %}){: style="max-width:70%;"}<br><br>
5. Select **Create API key**.
6. In the **API Keys** tab, copy the identifier next to the API key you created.
7. Paste the RelayState API Key into your IdP's RelayState (it may also appear as Relay State or Default Relay State) depending on your IdP.

> ### The user is stuck in a sign-in loop

Reviewer: ```suggestion
### The user is stuck in a sign-in loop between Okta and Braze
```

> If you're using Okta and the user is stuck in a sign-in loop (in other words, is cycling through SSO and Braze dashboard), you need to set the destination to `https://dashboard-dashboard-instance-here.braze.com`. 

Reviewer: ```suggestion
If a user can't sign in because they're stuck cycling between the Okta SSO and Braze dashboard, you need to go to $LOCATION and set the destination to `https://dashboard-dashboard-instance-here.braze.com`. 
```

be sure to replace `$LOCATION` with where the reader needs to go to set this.

Author: @paulzs Where does the user set the destination? Is it within Okta?

Reviewer 2: Yes, that would be set in Okta. I think it's called the "Single sign-on URL" or something like that.
